test/test_optimizers.f90

```diff
@@ -0,0 +1,150 @@
+program test_optimizers
+  use nf, only: dense, input, network
+  use nf_loss, only: quadratic
+  use iso_fortran_env, only: stderr => error_unit
+  use nf_optimizers, only: sgd, rmsprop
+  implicit none
+  type(network) :: net(5)
+  real, allocatable :: x(:), y(:)
+  real, allocatable :: ypred(:)
+  integer, parameter :: num_iterations = 1000
+  real :: tolerance = 1e-3
```

Comment:
Since this is only used in `check_convergence`, declare it there instead of here and make it a `parameter`.